tests/auto/metallicroughnesseffect/tst_metallicroughnesseffect.cpp

```diff
@@ -0,0 +1,222 @@
+        }
+
+        {
+            // WHEN
+            effect.setEmissiveMapEnabled(false);
+            effect.setEmissiveMapEnabled(true);
+            effect.setEmissiveMapEnabled(true);
+
+            // THEN
+            QCOMPARE(effect.isEmissiveMapEnabled(), true);
+            QCOMPARE(emissiveMapSpy.count(), 1);
+        }
+
+        {
+            // WHEN
+            effect.setUsingColorAttribute(false);
+            effect.setUsingColorAttribute(true);
+            effect.setUsingColorAttribute(true);
+
+            // THEN
+            QCOMPARE(effect.isUsingColorAttribute(), true);
+            QCOMPARE(usingColorAttributeSpy.count(), 1);
+        }
+
+        {
+            // WHEN
+            effect.setDoubleSided(false);
+            effect.setDoubleSided(true);
+            effect.setDoubleSided(true);
+
+            // THEN
+            QCOMPARE(effect.isDoubleSided(), true);
+            QCOMPARE(doubleSidedSpy.count(), 1);
+        }
+
+        {
+            // WHEN
+            effect.setUseSkinning(false);
+            effect.setUseSkinning(true);
+            effect.setUseSkinning(true);
+
+            // THEN
+            QCOMPARE(effect.useSkinning(), true);
+            QCOMPARE(useSkinningSpy.count(), 1);
+        }
+
+        {
+            // WHEN
+            effect.setOpaque(true);
+            effect.setOpaque(false);
+            effect.setOpaque(false);
+
+            // THEN
+            QCOMPARE(effect.isOpaque(), false);
+            QCOMPARE(opaqueSpy.count(), 1);
+        }
+
+        QCOMPARE(effect.isAlphaCutoffEnabled(), false);
+        {
+            // WHEN
+            effect.setAlphaCutoffEnabled(false);
+            effect.setAlphaCutoffEnabled(true);
+            effect.setAlphaCutoffEnabled(true);
+
+            // THEN
+            QCOMPARE(effect.isAlphaCutoffEnabled(), true);
+            QCOMPARE(alphaCutoffSpy.count(), 1);
+        }
+
+        {
+            // Workaround ToneMapping not being registered for QVariants
+            int emitCount = 0;
+            QObject::connect(&effect, &MetallicRoughnessEffect::toneMappingAlgorithmChanged, [&emitCount]() {
+                emitCount++;
+            });
+
+            // WHEN
+            effect.setToneMappingAlgorithm(MetallicRoughnessEffect::Reinhard);
+            effect.setToneMappingAlgorithm(MetallicRoughnessEffect::Filmic);
+            effect.setToneMappingAlgorithm(MetallicRoughnessEffect::Filmic);
+
+            // THEN
+            QCOMPARE(effect.toneMappingAlgorithm(), MetallicRoughnessEffect::Filmic);
+            QCOMPARE(emitCount, 1);
+        }
+
+        {
+            effect.setBrdfLUT(nullptr);
+
+            auto *texture2D = new Qt3DRender::QTexture2D();
+
+            effect.setBrdfLUT(texture2D);
+            effect.setBrdfLUT(texture2D);
+
+            QCOMPARE(effect.brdfLUT(), texture2D);
+            QCOMPARE(brdfLUTSpy.count(), 1);
+        }
+    }
+};
+
```

Comment:
This test is fine but it' s not testing for your new properties. You might have to write tst_MetallicRoughnessMaterial I fear

Reply:
Yes, I was adding that. Added to this patch